cluster-autoscaler/simulator/utilization/info.go

```diff
@@ -118,7 +118,14 @@ func CalculateUtilizationOfResource(nodeInfo *schedulerframework.NodeInfo, resou
 		if drain.IsPodLongTerminating(podInfo.Pod, currentTime) {
 			continue
 		}
-		for _, container := range podInfo.Pod.Spec.Containers {
+		// calculate all regular containers and restartable init containers requests
```

Comment:
The new calculation takes restartable initContainers (which is a new thing available in k8s 1.28+) into account, but not regular initContainers. We should be calculating the same way as kube-scheduler, the logic is here: https://github.com/kubernetes/kubernetes/blob/8509ab82b96caa2365552efa08c8ba8baf11c5ec/pkg/api/v1/resource/helpers.go#L50.

Roughly:
* The previous code just looks at a sum of container requests.
* The proposed code looks at the sum of container and restartable initContainer requests.
* The correct logic should take a max of the proposed code (sum of container and restartable initContainer requests), and the biggest initContainer requests. Now computing the effective initContainer request unfortunately requires iterating over them, because you have to also add all previous restartable initContainer requests.

The correct logic is explained in this KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/753-sidecar-containers#exposing-pod-resource-requirements. The KEP also mentions that the computed value should already be available in the `TotalResourcesRequested` field on the pod, but I don't think that's available yet.

Reply:
Great idea, somehow skipped my mind :sweat_smile:

LGTM, I'd just like to see the non-restartable initContainer test case.